### ISSUE TITLE
unable to compile (2015-08-24)

### ISSUE THREAD

**Issue by vn971** [servo/gaol#6]
Hi! Unfortunately, `gaol` does not compile using the current "stable" and "nightly" rustc.

Error with stable:

```
lib.rs:11:1: 11:21 error: #[feature] may not be used on the stable release channel
lib.rs:11 #![feature(convert)]
          ^~~~~~~~~~~~~~~~~~~~
```

Error with "rustc 1.4.0-nightly (10d69db0a 2015-08-23)":

```
tests/file-read-metadata.rs:65:76: 65:93 error: no method named `position_elem` found for type `[u8; 4096]` in the current scope
tests/file-read-metadata.rs:65             PathBuf::from(OsStr::from_bytes(&new_temp_path[..new_temp_path.position_elem(&0)
                                                                                                          ^~~~~~~~~~~~~~~~~
tests/file-read-metadata.rs:65:76: 65:93 help: items from traits can only be used if the trait is in scope; the following trait is implemented but not in scope, perhaps add a `use` for it:
tests/file-read-metadata.rs:65:76: 65:93 help: candidate #1: use `core::slice::SliceExt`
error: aborting due to previous error
```

**Comment by vn971** [comment]
What I found out:

* the code does compile using rustc-2015-08-01

* if I understand it correctly, `position_elem` method was removed without a "breaking_change" marker. Maybe because it was marked deprecated already, I don't know the "social" contract.

* the following issue discusses the `position_elem` method and the possibility to return it back: https://github.com/rust-lang/rust/issues/27517  (again, if I understood it correctly).
